boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

ArtifixPrime opened the pull request:
This has been done mainly by adding support for more peripherals and
changing pin numbering from the existing port for F446RE board.
Additionally, dts/arm/st/f4/stm32f4.dtsi has been modified by adding
USART3 configuration.
It has been tested to work with the provided Zephyr samples.

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

ArtifixPrime committed:
boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

This has been done mainly by adding support for more peripherals and
changing pin numbering from the existing port for F446RE board.
It has been tested to work with the provided Zephyr samples.

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

ArtifixPrime committed:
boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

This has been done mainly by adding support for more peripherals and
changing pin numbering from the existing port for F446RE board.
Additionally, dts/arm/st/f4/stm32f4.dtsi has been modified by adding
USART3 configuration.
It has been tested to work with the provided Zephyr samples.

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

hotfix: boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

Added missing license to board.cmake
Fixed doc image in index.rst
Added flash layout to nucleo_f446ze.dts

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

hotfix: boards: nucleo_f446ze: Added support for the Nucleo144 F446ZE board

doc: rst-class directive to index.rst

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

ArtifixPrime committed:
boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

This has been done mainly by adding support for more peripherals and
changing pin numbering from the existing port for F446RE board.
Additionally, dts/arm/st/f4/stm32f4.dtsi has been modified by adding
USART3 configuration.
It has been tested to work with the provided Zephyr samples.

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

bfrog commented:
This is missing the flash layout like the RE board did, which means mcumgr, mcuboot, and various other flash related samples will fail to work

ArtifixPrime committed:
hotfix: boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

Added missing license to board.cmake
Fixed doc image in index.rst
Added flash layout to nucleo_f446ze.dts

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

dbkinder commented on boards/arm/nucleo_f446ze/doc/index.rst:
```diff
@@ -0,0 +1,243 @@
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_3_TX : PD8
```
If you add this rst-class directive and a blank line before the list, it will display the list in multiple columns instead of a long list that runs down the page:
```
.. rst-class:: rst-columns

- UART_3_TX : PD8
- UART_3_RX : PD9
  etc.
```

ArtifixPrime replied:
Noted. I've never worked with the Sphinx documentation before, so thanks for the tip.

dbkinder reviewed (changes requested):
one recommended appearance tweak otherwise looks fine

ArtifixPrime committed:
hotfix: boards: nucleo_f446ze: Added support for the Nucleo144 F446ZE board

doc: rst-class directive to index.rst

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

erwango commented on dts/arm/st/f4/stm32f4.dtsi:
```diff
@@ -147,6 +147,15 @@
 			label = "UART_2";
 		};
 
+		usart3: serial@40004800 {
```
Also, please do it in a dedicated commit

ArtifixPrime committed:
Squash into first commit added_support_for_nucleo144_f446ze_board

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

ArtifixPrime committed:
boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

Fixed errors after squashing

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

ArtifixPrime committed:
boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

Moved usart3 configuration from stm32f4.dtsi to stm32f446.dtsi

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

ArtifixPrime committed:
boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

Added UART3 configuration for STM32F446 chips.

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

ArtifixPrime committed:
Squash branch 'added_support_for_nucleo144_f446ze_board' of https://github.com/ArtifixPrime/zephyr into added_support_for_nucleo144_f446ze_board

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

erwango commented on boards/arm/nucleo_f446ze/nucleo_f446ze.dts:
```diff
@@ -0,0 +1,174 @@
+#include <st/f4/stm32f446Xe.dtsi>
+#include "arduino_r3_connector.dtsi"
+
+<<<<<<< HEAD
```
Please resolve merge conflicts.

ArtifixPrime replied:
Could you please help me out a bit?
I've checked all of the files and on my local machine git status returns:
"nothing to commit, working tree clean"

Shippable still fails some checks and prints in the console to resolve merge conflicts.

Sorry, I know this may be a silly request but I've never really worked with git before.

erwango replied:
> Sorry, I know this may be a silly request but I've never really worked with git before.

@ArtifixPrime, that's ok, there's a first time for everything. I won't be able to help today, but Iwill come back to you this week.

ArtifixPrime replied:
Alright, thank you so much! 
Sorry for replying this late, I am also busy with my own work, so whenever you'll find the time

ArtifixPrime committed:
boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

Squash commit

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

ArtifixPrime committed:
boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

Attempt at fixing errors made while squashing.
Added a redundant space character in readme.

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

erwango commented:
@ArtifixPrime, I'll try to provide feedback this week. Sorry for the delay.

erwango commented:
@ArtifixPrime, I had a look, and maybe the easiest way will be the following:
-In your current branch, copy the board directory on a location outside zephyr (the usart3 addition in stm32f4.dtsi is now in zephyr so you don't need it anymore)
-Fetch a latest zephyr version and create a **new** branch on top of the master
-Copy the board directory back in zephyr repo
-Push this new branch 

We'll continue work from there.

galak commented:
@ArtifixPrime any plans to update?

nashif commented:
stale, closing. Reopen when there are updates...

towen commented:
it's a shame this never made it in. I would really like to evaluate zephyr on this board 😢 
I have tried copying the board definition files out of ArtifixPrime's repo but I get errors compiling the device tree:
```
$ west build -p auto -b nucleo_f446ze samples/subsys/usb/cdc_acm_composite
-- west build: generating a build system
Including boilerplate (Zephyr base): /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/samples/subsys/usb/cdc_acm_composite
-- Zephyr version: 2.6.99 (/home/tom/Git/stm32_rtos_test/zephyr_test/zephyr), build: zephyr-v2.6.0-533-gc68054c9e519
-- Found Python3: /usr/bin/python3.9 (found suitable exact version "3.9.2") found components: Interpreter 
-- Found west (found suitable version "0.11.0", minimum required is "0.7.1")
-- Board: nucleo_f446ze
-- Cache files will be written to: /home/tom/.cache/zephyr
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Using toolchain: zephyr 0.12.4 (/opt/zephyr-sdk-0.12.4)
-- Found dtc: /opt/zephyr-sdk-0.12.4/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/boards/arm/nucleo_f446ze/nucleo_f446ze.dts
Lexical error: nucleo_f446ze.dts.pre.tmp:632.21-23 Bad integer literal '09'
Lexical error: nucleo_f446ze.dts.pre.tmp:634.9-11 Bad integer literal '08'
Lexical error: nucleo_f446ze.dts.pre.tmp:635.9-11 Bad integer literal '09'
Lexical error: nucleo_f446ze.dts.pre.tmp:638.21-23 Bad integer literal '09'
Lexical error: nucleo_f446ze.dts.pre.tmp:646.21-23 Bad integer literal '09'
Lexical error: nucleo_f446ze.dts.pre.tmp:647.21-23 Bad integer literal '08'
Error: nucleo_f446ze.dts.pre.tmp:665.22-23 syntax error
FATAL ERROR: Unable to parse input tree
CMake Error at /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/cmake/dts.cmake:214 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/cmake/app/boilerplate.cmake:571 (include)
  /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
```

erwango commented:
> it's a shame this never made it in. I would really like to evaluate zephyr on this board 
> I have tried copying the board definition files out of ArtifixPrime's repo but I get errors compiling the device tree:
> 

@towen This PR is pretty old and you will face lot of issues if trying to use on v2.6.0
I rather suggest you try to start from existing nucleo_f446re and adapt from there.
